api/src/main/java/org/openmrs/api/context/Daemon.java

```diff
@@ -271,7 +271,7 @@ public void run() {
 				isDaemonThread.set(true);
 				try {
 					Context.openSession();
-					runnable.run();
+                    new Thread(runnable).start();
```

Comment:
This is actually intended as you can see the start method is called below. Please revert it. We need to annotate the run method (line 270) with:

@SuppressWarnings("squid:3AS1217") //We intentionally do not start a new thread yet, rather wrap the run call in a session.